pkgs/by-name/cq/cq/package.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, fetchFromGitHub
+, fetchurl
+, buildGraalvmNativeImage
+, graalvmCEPackages
+}:
+
+buildGraalvmNativeImage rec {
+  pname = "cq";
+  version = "2024.02.13-23.14";
+
+  src = fetchurl {
+    url = "https://github.com/markus-wa/cq/releases/download/${version}/cq.jar";
+    sha256 = "sha256-JQp9fBR7jhenXcXSVMFsWosKUsjNxWVU+QnTz0X+dwk=";
+  };
+
+  build-src = fetchFromGitHub {
+    owner = "markus-wa";
+    repo = "cq";
+    rev = version;
+    hash = "sha256-Ir4IrUucouKz3gPPrpCigyb+utE8t0I1Hzb/Rb88haM=";
+  };
+
+  graalvmDrv = graalvmCEPackages.graalvm-ce;
+
+  executable = "cq";
+
+  extraNativeImageBuildArgs = [
```

Comment:
Could you add a comment to clarify why that options is needed? Or at least a link to the upstream build script, that could help us in the future.
Other than that LGTM :)

Reply:
Done. As much as I want to stop using the prebuilt jar, I can't figure how to build the clojure part by hand.

Reply:
We don't have any helpers on nixpkgs to build a clojure application from source, for now the common approach for clojure (and many other jvm applications), is to use the prebuilt uberjar. 
If you want to build cq without relaying on the prebuilt jar, you could use https://github.com/jlesquembre/clj-nix (I'm the author).